Sources/ClientRuntime/Networking/Http/Middlewares/DeserializeMiddleware.swift

```diff
@@ -18,7 +18,9 @@ public struct DeserializeMiddleware<Output: HttpResponseBinding,
             let response = try await next.handle(context: context, input: input) // call handler to get http response
             var copiedResponse = response
             if (200..<300).contains(response.httpResponse.statusCode.rawValue) {
-                let output = try Output(httpResponse: copiedResponse.httpResponse, decoder: decoder)
+                let output = try Output(httpResponse: copiedResponse.httpResponse,
+                                        decoder: decoder,
+                                        messageDecoder: context.getMessageDecoder())
```

Comment:
Only event streams require the message decoder/encoder, it'd be great if we didn't need to do this for all. I almost think that event-streams should have its own serde middlewares, especially since they have their own behavior (serialize headers -> sign request -> serialize chunk -> sign chunk -> serialize chunk -> sign chunk etc.....)